File: LEETCODE/Arrays/1.1 Two Sum/main.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        HashMap<Integer,Integer>H = new HashMap<>();
        for(int i = 0 ; i < nums.length; i++)
        {
            int comp = target-nums[i];
            if(H.containsKey(comp))
            {
                result[0] = H.get(comp);
                result[1] = i;
            }
            else
            {
                H.put(nums[i],i);
            }
        }
        return result;
    }
}

File: LEETCODE/Arrays/1.1 Two Sum/main.js
var twoSum = function(nums, target) {
    let map = new Map();
    for(let i = 0 ; i < nums.length; i++){
        
        let n1 = nums[i];
        let n2 = target - nums[i];
        
        if(map.has(n2)){
            return [i,map.get(n2)];
        }
        
        map.set(n1,i);
    }
};
